server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -270,9 +270,13 @@ public SearchRequest(StreamInput in) throws IOException {
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
+        writeTo(out, false);
+    }
+
+    public void writeTo(StreamOutput out, boolean skipIndices) throws IOException {
```

Comment:
This is admittedly a fairly lazy solution but benchmarking that match_all against 50k shards across 5 nodes would burn 2% of coordinating node CPU (adding considerably noise to transport thread performance) on serializing the indices list. As a matter of principle I found it useful to do away with an obvious  non-scaling thing like that in this first step.

Reply:
this is the array of indices that the search request targets, isn't it? Why does the number of shards matter in how expensive is it to serialize it, especially for the batched codepath? I would intuitively expect that we rather need to get rid of it for the fan out case, like probably many other things that are duplicated which batching addresses as a whole?

Reply:
I am forwarding the `SearchRequest` to each node. The number of shards doesn't matter directly but the number of indices does. As an extreme example from the many-shards benchmark: if I target 5 nodes, 50k indices and it's 10k shards per node then I'm sending 50k indices to each of the 5 nodes needlessly. That's 250k strings and many MB of needless serialization/traffic given that the indices array is never used in the fanout logic.
This could be dealt with nicer by not using `SearchRequest` for the fanout but that requires a considerable refactoring of the query phase logic and this array seems to be the only big overhead from just reusing `SearchRequest` so I figured I'd just go for the cheap option :).

Reply:
Ah this was hidden somehow unless you open the files view on the PR :) 
I think my suggestion would be for an immediate follow-up that just removes the use of `SearchRequest` here instead of making more adjustments to it. I made this adjustment as the smallest possible stop-gap solution for this PR with the only goal of avoiding the massive list getting sent around the cluster. But really, we just shouldn't be sending the full `SearchRequest` or use that class at all. Unfortunately, doing so requires about ~300 lines of changes so I backed that out of here. Maybe just leave this for an instant follow-up to keep things simple?

Reply:
adjusted the code to ship `["*", "-*"]` here for now as discussed in Slack :)